cypress-tests/cypress/support/commands.js

```diff
@@ -86,36 +85,30 @@ Cypress.Commands.add("apiKeyCreateTest", (apiKeyCreateBody, globalState) => {
 Cypress.Commands.add("createConnectorCallTest", (createConnectorBody, globalState) => {
   const merchantId = globalState.get("merchantId");
   createConnectorBody.connector_name = globalState.get("connectorId");
-  const authDetails = getValueByKey(ConnectorAuthDetails, globalState.get("connectorId"));
-  createConnectorBody.connector_account_details = authDetails;
-  cy.request({
-    method: "POST",
-    url: `${globalState.get("baseUrl")}/account/${merchantId}/connectors`,
-    headers: {
-      "Content-Type": "application/json",
-      Accept: "application/json",
-      "api-key": globalState.get("adminApiKey"),
-    },
-    body: createConnectorBody,
-    failOnStatusCode: false
-  }).then((response) => {
-    // Handle the response as needed
-    const xRequestId = response.headers['x-request-id'];
-    if (xRequestId) {
-      cy.task('cli_log', "x-request-id ->> " + xRequestId);
-    } else {
-      cy.task('cli_log', "x-request-id is not available in the response headers");
-    }
-
-    if (response.status === 200) {
-      expect(globalState.get("connectorId")).to.equal(response.body.connector_name);
-    }
-    else {
-      cy.task('cli_log', "response status ->> " + JSON.stringify(response.status));
-      cy.task('cli_log', "res body ->> " + JSON.stringify(response.body));
-    }
-
-    
+  // readFile is used to read the contents of the file and it always returns a promise ([Object Object]) due to its asynchronous nature
+  // it is best to use then() to handle the response within the same block of code
+  cy.readFile(globalState.get("connectorAuthFilePath")).then((jsonContent) => {
+    const authDetails = getValueByKey(JSON.stringify(jsonContent), globalState.get("connectorId"));
+    createConnectorBody.connector_account_details = authDetails;
+    cy.request({
+      method: "POST",
+      url: `${globalState.get("baseUrl")}/account/${merchantId}/connectors`,
+      headers: {
+        "Content-Type": "application/json",
+        Accept: "application/json",
+        "api-key": globalState.get("adminApiKey"),
+      },
+      body: createConnectorBody,
+    }).then((response) => {
+      logRequestId(response.headers['x-request-id']);
+
+      if (response.status === 200) {
+        expect(globalState.get("connectorId")).to.equal(response.body.connector_name);
+      } else {
+        cy.task('cli_log', "response status -> " + JSON.stringify(response.status));
+        cy.task('cli_log', "res body -> " + JSON.stringify(response.body));
+      }
+    });
```

Comment:
`ConnectorAuthDetails` in [line 29](https://github.com/juspay/hyperswitch/blob/705e82779a2b7bfd0cb1cd856b4a760d487cd8c5/cypress-tests/cypress/support/commands.js#L29) reads the content from the file and stores within that variable.

`getValueByKey()` takes the content of the file and the key (the connector name) and returns the respective connector auth details.

Since we now read from the env, we do not have any method to store the content. Cypress natively supports `cy.readFile` which is again, asynchronous in nature, returns an Object (promise). We can only parse that Object within `.then()` and read it.

With that, it makes it necessary for us to move the entire flow within `cy.readFile()` [Docs: https://docs.cypress.io/api/commands/readfile]